lib/controller.rb

```diff
@@ -6,6 +6,18 @@
 
 class WeekThreeAssessment < Sinatra::Base
 
-  # Your code will go here.
+  get '/' do
+    erb :index
+  end
 
+  def user
+    User.new
+  end
+
+  get '/name/:name' do
+    if user.valid?
+      user.save
+    end
+  end
+  DataMapper.finalize
```

Comment:
DataMapper commands are usually put on top of the file but in this app it's extracted to a separate file that is required by the controller: /lib/helpers/data_mapper.rb